Add feature doc template (#4355)
ovn-org/ovn-kubernetes: docs/features/template.md

# Name of the Feature

* Add diagrams wherever possible to explain things better

> * Keep the end user in mind as target audience when you write thisdocumentation; not an ovn expert

Reviewer: ```suggestion
* Keep the end user in mind as target audience when you write this documentation; not an ovn expert
```

Author: done

### User-Stories/Use-Cases

NOTE: If you had an enhancement proposal written for this feature
then copy the user stories over from there OR provide a link
to that enhancement proposal.

(What new user-stories/use-cases does this feature introduce?)

A user story should typically have a summary structured this way:

1. **As a** [user concerned by the story]
2. **I want** [goal of the story]
3. **so that** [reason for the story]

The “so that” part is optional if more details are provided in the description.
A story can also be supplemented with examples, diagrams, or additional notes.

e.g

Story 1: Deny traffic at a cluster level

As a cluster admin, I want to apply non-overridable deny rules to certain pod(s)
and(or) Namespace(s) that isolate the selected resources from all other cluster
internal traffic.

For Example: The admin wishes to protect a sensitive namespace by applying an
AdminNetworkPolicy which denies ingress from all other in-cluster resources
for all ports and protocols.

## How to enable this feature on an OVN Kubernetes cluster?

Did you write new configs or knobs that need to be turned on
for this feature? If so educate end users on those details.

> How can users disabled this feature if they don't want it?

Reviewer: ```suggestion
How can users disable this feature if they don't want it?
```

Author: done

## Workflow Description

Explain how the user will use the feature. Be detailed and explicit.
Describe all of the actors, their roles, and the APIs or interfaces
involved. Define a starting state and then list the steps that the
user would need to go through to trigger the feature described in the
enhancement. **A detailed architectural diagram is a must to showcase
how this feature works in our CNI.** Remember that a picture can speak
a thousand words.

> ### Implementation Details

Reviewer 2: ```suggestion
## Implementation Details
```

Author: done.

Use the following sub-sections to explain more implementation details in a top down fashion

### User facing API Changes

If any API changes were done, they must be outlined here and
look at the developer docs on how to generate API reference
documentation for this feature. Provide a link to that API
reference web page from this spot.
Also add details on where the CRD lives; example is it in
kubernetes or kubernetes-sigs or network-plumbing-working-group?

> ### OVN Kubernetes Implementation Details

Reviewer: Should it be `OVN-Kubernetes` instead?

Reviewer 3: yes, using OVN-Kubernetes, will be better.

Author: hmm the recommendation I got from PR team/docs team when I wrote https://www.redhat.com/en/blog/how-the-new-ovn-kubernetes-architecture-helps-with-scaling-openshift was that we must call it either as `ovn-kubernetes` repo name OR `OVN Kubernetes`; hyphenating is not correct, we can make it `OVNKubernetes` as well ?

Author: going with OVN-Kubernetes, resolving issue

data plane to make this happen? Note differences if any for local
gateway versus shared gateway and default mode versus interconnect
mode. **A detailed OVN Kubernetes networking topology diagram is a must
to showcase how this feature works in our CNI**. Remember that a picture
can speak a thousand words. Give details on how the above API is watched
and converted into OVN objects by OVN Kubernetes. If there are changes
done on host-networking level like adding new routes they must
be highlighted.

#### OVN Constructs created in the databases

What were the OVN Objects used? Give snippets and details on
the various database objects that this feature creates in the
OVN NBDB and SBDB. Keep in mind goal is to educate users how this
feature was implemented using OVN in this section. You can go as deep
as you want like mentioning how the pipelines get executed
for your feature.

#### OVS Flows generated

Provide details on specific changes done to br-ex, br-int or how
the relevant openflows generated by this feature look like.
Keep in mind goal is to educate users how this
feature was implemented using OVS in this section.

## Future Items

Add a bullet list of future plans if any

> 

Author: add troubleshooting section

Author: also best practices and scale information
